tests/test_cli_compile.py

```diff
@@ -839,3 +839,27 @@ def test_empty_input_file_no_header(runner, empty_input_pkg, prior_output_pkg):
 
     with open("requirements.txt", "r") as req_txt:
         assert req_txt.read().strip() == ""
+
+
+def test_pip_compile_upgrade_package(pip_conf, runner):
```

Comment:
```suggestion
def test_upgrade_package_doesnt_remove_annotation(pip_conf, runner):
```

This test name could be more descriptive for my taste.